src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -2273,6 +2832,16 @@ ucs_status_t uct_ib_mlx5_devx_mem_attach(uct_md_h uct_md,
     hdr       = UCT_IB_MLX5DV_ADDR_OF(create_alias_obj_in, in, hdr);
     alias_ctx = UCT_IB_MLX5DV_ADDR_OF(create_alias_obj_in, in, alias_ctx);
 
+    /* If received mkey is UMR key, find alias in the UMR hash map */
+    if (uct_ib_mlx5_devx_umr_mkey_is_xgvmi(packed_mkey->lkey)) {
+        status = uct_ib_mlx5_devx_umr_mkey_hash_find(md, packed_mkey,
```

Comment:
major: maybe the hash table could hold full memh as a value, instead of just lkey+cross_mr?

Reply:
Mmm, to me this idea seems controversial to the original design.
My understanding is: we keep the hash map of mkey aliases on the importer side, so that we can reuse them with different memory handles. This is the principal use case for UMR approach, also called "temporary workflow" or "non-persistent workflow". We gain performance because we reuse the same UMR mkey on the exporter side, and corresponding mkey alias on the importer side for *different* memory handles.
Now if we bind alias to memh, that would break the design I'm afraid.

Reply:
Yes, I see your point, but I'm not sure if it's simpler.
My concern is related to the lifetime and usage of those memory handles. So far the typical workflow on the daemon side for those memory handles is:
```
in INIT or SEND_REQ message handler:
    imported_memh = ucp_mem_map(buf); // internally creates new memh in mem_attach
    ...
    ucp_am_send_nbx(imported_memh); // send request to the receiver daemon
    ...
    ucp_mem_unmap(imported_memh); // cleanup the resource
```
With "regular workflow" we allocate memh once in INIT handler, and deallocate in FIN handler. But with "temporary workflow" we allocate on SEND_REQ, and deallocate once receive ACK from the peer daemon. So it's the user who controls the lifetime of those memory handles with API.
Now you propose to make those memory handles persistent. Ok, but then what happens if client destroys the handle with `ucp_mem_unmap` call? My assumption is that we need to destroy the memh object, and keep persistent only the alias mkey - to be reused in the future. This is current impl.
But your approach is also doable: we just need to check in `ucp_mem_unmap` call if memh being destroyed is present in the hash map, and if it is, then do nothing. Then this persistent memh can be destroyed only after removing from hash map. Is it what you are proposing?
Is it simpler? I don't know, it changes a bit the API semantics.
I'm not sure if there are other consequences of those persistent memory handles - like if you call `ucp_mem_map` several times, you always get the same object returned. It can be ambiguous.

IMO current design is simpler, as it does not change the semantics of existing stuff. Please let me know your thoughts